conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/generate/ConjureUndertowEndpointsGenerator.java

```diff
@@ -159,7 +159,8 @@ private static TypeSpec endpoint(ServiceDefinition service, EndpointDefinition e
         endpoint.arguments().forEach(def -> def.paramType().match(new Cases<Void>() {
             @Override
             public Void body(CodeBlock deserializerFactory, String deserializerFieldName) {
-                TypeName requestBodyType = def.argType().match(ArgTypeTypeName.INSTANCE);
+                TypeName requestBodyType =
+                        def.argType().match(ArgTypeTypeName.INSTANCE).box();
```

Comment:
This is the fix, the rest is test coverage

Reply:
Random question, but why would this compile before hand? Under what cases does box change the result type?

Reply:
`int` -> `Integer`?

Reply:
It failed because it tried to create a field of type `Deserializer<int>` where we need `Deserializer<Integer>`